gemd/units/impl.py

```diff
@@ -285,36 +252,6 @@ def _format_clean(unit, registry, **options):
 
 
 @functools.lru_cache(maxsize=1024)
-def _parse_units(units: str) -> Unit:
-    """
-    Parse a string or Unit into a standard string representation of the unit.
-
-    Parameters
-    ----------
-    units: Union[str, Unit, None]
-        The string or Unit representation of the object we wish to display
-
-    Returns
-    -------
-    [Union[str, Unit, None]]
-        The representation; note that the same type that was passed is returned
-
-    """
-    # TODO: parse_units has a bug resolved in 0.19, but 3.7 only supports up to 0.18
-    parsed = _REGISTRY(units)
-    try:
-        magnitude = parsed.magnitude
-        result = parsed.units
-    except AttributeError:  # It was non-dimensional
-        magnitude = parsed
-        result = _REGISTRY("").u
-    if magnitude == 0.0:
-        raise ValueError(f"Unit expression had a zero scaling factor. {units}")
-    if magnitude != 1:
-        raise ValueError(f"Unit expression cannot have a leading scaling factor. {units}")
-    return result
-
-
```

Comment:
This is no longer required because dropping 0.18 support drops the buggy code that was causing trouble